src/utils/time.cc

```diff
@@ -73,4 +76,30 @@ cr_timestamp_nsec_t GetUnixTimestampNsec() {
     return static_cast<cr_timestamp_nsec_t>(duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count());
 }
 
+std::string GetCurrentUtcDate() {
+    return GetUtcDate(std::time(nullptr));
+}
+
+std::string GetUtcDate(const std::time_t ts) {
+    return fmt::format("{:%Y-%m-%d}", fmt::gmtime(ts));
+}
+
+std::string GetCurrentUtcHour() {
+    return GetUtcHour(std::time(nullptr));
+}
+
+std::string GetUtcHour(const std::time_t ts) {
+    return fmt::format("{:%Y-%m-%d %H}", fmt::gmtime(ts));
+}
+
+bool IsSameDay(const std::time_t x, const std::time_t y) noexcept {
```

Comment:
如果你只是要判断utc同天，那`system_clock`就够了

> system_clock measures Unix Time (i.e., time since 00:00:00 Coordinated Universal Time (UTC), Thursday, 1 January 1970, not counting leap seconds).

Reply:
因为
- C++-style vs C-style
- 不用自行转换。可能你注意到了，自行转换的部分我们只用了ratio的一测，这是由于我们知道另一侧一定是1。写全了也很丑的
- 这里你定义两个临时变量再比较就好看一些了

Reply:
就看你上面的例子，你用了`duration::period`做运算。而根据https://en.cppreference.com/w/cpp/chrono/duration

> period |    typename Period::type (since C++17), a [std::ratio](https://en.cppreference.com/w/cpp/numeric/ratio/ratio) representing the tick period (i.e. the number of second's fractions per tick)

简单说就是period表示了这个单位和“秒”的比例关系。

而period是有理数`std::ratio<Num, Denom>`。num是分子，denom是分母。那你这里既可以是`86400 / 1`，也可以是`172800 / 2`。

当然了，`chrono::duration`自己说了，days的period就是`std::ratio<86400>`，ratio也说了，你用`::num` `::den`拿到的分子分母都是最简形式的，所以你上面的写法是没问题的。但对于reviewer来说，判断这段的正确性需要他们熟悉标准里的这些规定，而不是从代码上看不言自明的。所以我觉得虽然duration_cast虽然看着多一点，但是不需要reviewer知道这些规定细节也能判断这段没写错。